delirium/dns/cache.py

```diff
@@ -25,18 +28,73 @@
 QUERY_GET_BY_NAME = "SELECT * FROM cache WHERE name = :name AND expired = :expired;"
 QUERY_UPDATE_DATE_BY_ID = "UPDATE cache SET date = :date WHERE id = :id;"
 
+def get_addr_range(value):
+    """Converts a <ip>-<ip> string to integers for random.randrange()"""
+
+    values = value.split('-')
+    s_int = struct.unpack('!L', socket.inet_aton(values[0]))[0]
+    e_int = struct.unpack('!L', socket.inet_aton(values[1]))[0]
+    return s_int, e_int
+
+
+def n_generator(start, end):
+    for i in itertools.cycle(range(start, end + 1)):
+        yield i
+
+class CacheObject:
```

Comment:
I'd like to see this abstract class and `CacheDatabase` combined into one class.  As we're only supporting one cache-type now, there really isn't a need for an abstract class.  I'll explain why on our call tonight.